### PR TITLE
6218162: DefaultTableColumnModel.getColumn() method should mention ArrayIndexOutOfBoundsException

### PR DESCRIPTION
**Pull request by prsadhuk**
If invalid ie 0 or > getColumnCount() index is passed to DefaultTableColumnModel.getColumn() then it returns AIOBE in current implementation which should be documented in the spec. Fixed the spec to mention the exception.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed
- [x] Change requires a CSR request to be approved

### Issues
 * [JDK-6218162](https://bugs.openjdk.java.net/browse/JDK-6218162): DefaultTableColumnModel.getColumn() method should mention ArrayIndexOutOfBoundsException
 * [JDK-8282378](https://bugs.openjdk.java.net/browse/JDK-8282378): DefaultTableColumnModel.getColumn() method should mention ArrayIndexOutOfBoundsException (**CSR**)

### Reviewers
 * [Alexey Ivanov](https://openjdk.java.net/census#aivanov) (@aivanov-jdk - **Reviewer**)
 * [Phil Race](https://openjdk.java.net/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7587/head:pull/7587` \
`$ git checkout pull/7587`

Update a local copy of the PR: \
`$ git checkout pull/7587` \
`$ git pull https://git.openjdk.java.net/jdk pull/7587/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7587`

View PR using the GUI difftool: \
`$ git pr show -t 7587`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7587.diff">https://git.openjdk.java.net/jdk/pull/7587.diff</a>

</details>

### CODE REVIEW

File: src/java.desktop/share/classes/javax/swing/table/DefaultTableColumnModel.java [hunk 1]
```diff
@@ -290,6 +290,8 @@ public int getColumnIndex(Object identifier) {
      * @param   columnIndex     the index of the column desired
      * @return  the <code>TableColumn</code> object for the column
      *                          at <code>columnIndex</code>
+     * @exception  ArrayIndexOutOfBoundsException if <code>columnIndex</code>
+     *             is not in valid range (< 0 or >= getColumnCount())
```
Comment (aivanov-jdk) [inline src/java.desktop/share/classes/javax/swing/table/DefaultTableColumnModel.java:294]:
```suggestion
     * @exception  ArrayIndexOutOfBoundsException if <code>columnIndex</code>
     *             is not in the valid range (< 0 or >= getColumnCount())
```
I find it confusing: the expression in parenthesis seems to clarify the meaning of *the valid range* but doesn't.

I think it's better to make it clearer:
```suggestion
     * @exception  ArrayIndexOutOfBoundsException if <code>columnIndex</code>
     *             is out of range (<code>columnIndex < 0 || columnIndex >= getColumnCount()</code>)
```

**Review by aivanov-jdk — CHANGES_REQUESTED** [review]
This change probably requires a CSR.

**Review by aivanov-jdk — APPROVED** [review]
I believe the copyright year shall be updated too.

Other than that, it looks good.

File: src/java.desktop/share/classes/javax/swing/table/DefaultTableColumnModel.java [hunk 2]
```diff
@@ -290,6 +290,9 @@ public int getColumnIndex(Object identifier) {
      * @param   columnIndex     the index of the column desired
      * @return  the <code>TableColumn</code> object for the column
      *                          at <code>columnIndex</code>
+     * @exception  ArrayIndexOutOfBoundsException if <code>columnIndex</code>
+     *             is out of range:
```
Comment (prrace) [inline src/java.desktop/share/classes/javax/swing/table/DefaultTableColumnModel.java:294]: As Joe Darcy pointed out, we use @throws these days.

Comment (prsadhuk) [reply src/java.desktop/share/classes/javax/swing/table/DefaultTableColumnModel.java:294]: I was aware of that but this file uses @exception throughout, so to be consistent, I used that..But if it is recommended to use @throws even at the cost of being inconsistent within the file, I would use that.

**Review by aivanov-jdk — APPROVED** [review]

**Review by prrace — APPROVED** [review]

**Pull request closed without merging**